hdk-rust/src/api.rs

```diff
@@ -376,3 +406,680 @@ pub fn close_bundle(_action: BundleOnClose) -> Result<(), RibosomeError> {
     // FIXME
     Err(RibosomeError::FunctionNotImplemented)
 }
+//--------------------------------------------------------------------------------------------------
+// UNIT TESTS
+//--------------------------------------------------------------------------------------------------
+
+/// Unit tests
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    /**
+     * Ribosome error handling unit tests
+     */
+
+    #[test]
+    /// test that we can convert an error to a string
+    fn test_to_string() {
+        let err = RibosomeError::FunctionNotImplemented.to_string();
+        assert_eq!(r#"FunctionNotImplemented"#, err)
+    }
+
+    #[test]
+    /// test that we can get the description for an error
+    fn test_description() {
+        let err = RibosomeError::FunctionNotImplemented;
+        assert_eq!("Function not implemented", err.description())
+    }
+
+    /**
+     * property() unit tests
+     */
+
+    #[test]
+    /// test that property() returns HashNotFound error for null key
+    fn test_property_invalid() {
+        // check whether function implemented
+        let result = property("Name");
+        assert_ne!(Some(RibosomeError::FunctionNotImplemented), result.err());
+
+        // test empty property key parameter
+        let result = property("");
+        assert_ne!(Some(RibosomeError::HashNotFound), result.err());
+
+        // test unknown property key parameter
+        assert_eq!(
+            r#"HashNotFound"#,
+            property("unknown").err().unwrap().to_string()
+        );
+    }
+
+    #[test]
+    /// test that property() returns value for known key
+    fn test_property_valid() {
+        // check whether function implemented
+        let result = property("Name");
+        assert_ne!(Some(RibosomeError::FunctionNotImplemented), result.err());
+
+        /*** FIXME
+        // test known property key parameter
+        let result = property("Name");
+        assert!(result.is_ok())
+        ***/
+    }
+
+    /**
+     * make_hash() unit tests
+     */
+
+    #[test]
+    /// test that make_hash() returns value for array entry data
+    fn test_make_hash_invalid() {
+        // check whether function implemented
+        let result = make_hash("", json!("test_data"));
+        assert_ne!(Some(RibosomeError::FunctionNotImplemented), result.err());
+
+        // test empty entry type parameter
+        // TODO: is this the right error?
+        let result = make_hash("", json!("test_data"));
+        assert!(result.is_err());
+        assert_eq!(Some(RibosomeError::HashNotFound), result.err());
```

Comment:
I don't understand how this test can pass.. (?!)
`make_hash()` is returning `Err(RibosomeError::FunctionNotImplemented)` above.

Are these test not run be the CI?